modules/lvgl/input/lvgl_common_input.c

```diff
@@ -52,3 +55,21 @@ int lvgl_input_register_driver(lv_indev_type_t indev_type, const struct device *
 
 	return 0;
 }
+
+#define LV_DEV_INIT(node_id, init_fn, ret) ret |= init_fn(DEVICE_DT_GET(node_id));
+
+int lvgl_init_input_devices(void)
+{
+	int ret = 0;
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_pointer_input, LV_DEV_INIT,
+				     lvgl_pointer_input_init, ret)
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_button_input, LV_DEV_INIT, lvgl_button_input_init,
+				     ret)
+
+	DT_FOREACH_STATUS_OKAY_VARGS(zephyr_lvgl_encoder_input, LV_DEV_INIT,
+				     lvgl_encoder_input_init, ret)
+
+	return ret;
+}
```

Comment:
note that with iterable sections you can gather all init calls in a single array and call them in a loop. wouldn't it be a better option?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/61986 would be best, you just find all the struct devices and call `dev->init()` on them.

Iterable section would be cool but I'm wondering if it's worth setting that up considering that it'll be torn down once the one above gets IMO. What do you think?

Reply:
even if we have dev->init, I'd like to implement deferred init first (it likely will imply managing dependencies properly, even if in this case it does not apply)

Reply:
ok so you think iterable section for this in the meantime? I'm happy with either, bit of a hassle to set up the section but then it's nice that the init functions could stay static. @faxe1008 what do you think?

Reply:
@gmarull I'd love that but how do I associate the devices init function in a way so that they are not called during `POST_KERNEL`  "automatically"?

Reply:
btw, we should think of these devices as input devices that have a custom API call to perform LVGL gluing. Could that call be common (ie single API call that `return lvgl_input_register_driver(x, dev);` where `x` is, e.g. `LV_INDEV_TYPE_POINTER` (stored in dev->config)?

Reply:
I realize this will insert 3rd party module glue code in include/zephyr/linker/common-rom/common-rom-misc.ld, which is not ideal (IIRC this is due to a bug in xtensa on iterable sections, because they could be defined in an ld file placed wherever you want).

Reply:
Is it a problem? Would go behind a #if guard I imagine

Reply:
yeah, just that glue code is usually restricted to `modules/`

Reply:
Ok, I'll press the green button and say this is fine as is. :-)

Reply:
...apart from the undefined reference in CI...

Reply:
Sure would be possible but I'd rather pick the actual init methods, since it also cares about setting up device specific configs.  Do I understand correctly that you suggest declaring a structure like:
```
struct lvgl_pseudo_device {
       const struct device* dev;
       init_fn fn;
};
```
And store those in a new iterable section?